tools/cold-backup/README.md

```diff
@@ -0,0 +1,85 @@
+
+If you think you do not need a backup strategy for Kafka as it has embedded data replication,
+then try to immagine a misconfiguration/bug/security-breach deleting all your data. For hot/online
+backups, you should look at *MirrorMaker2* to sync with a remote cluster, but this comes with
+additional complexities and required resources.
+
+To run the script you must be logged in as *cluster-admin* user. The script will stop the operator
+and the whole cluster for the entire duration of the backup procedure. If you have a single cluster
+wide operator, then you need to manually scale it down before starting the backup.
+
+The final archive contains an *env* file with the operator's version that you need to deploy *after*
+the restore ahs finished. Only local file system is supported, consumer group offsets are included,
+but not KafkaConnect custom images, that are usually hosted on an external registry.
+
+## Requirements
+- bash 5+ (GNU)
+- kubectl 1.20+ (Kubernetes)
+- tar 1.33+ (GNU)
+- yq 4.5+ (YAML processor)
+- zip 3+ (Info-ZIP)
+- unzip 6+ (Info-ZIP)
+- enough disk space
+
+## Test procedure
+```sh
+# deploy a test cluster
+kubectl create namespace myproject
+kubectl config set-context --current --namespace="myproject"
+kubectl create -f ./install/cluster-operator
+kubectl create -f ./examples/kafka/kafka-persistent.yaml
+kubectl create -f ./examples/topic/kafka-topic.yaml
+kubectl create cm custom-cm --from-literal=foo=bar
+
+# send 100000 messages and consume them
+kubectl run kafka-producer-perf-test -it \
+    --image="quay.io/strimzi/kafka:latest-kafka-2.6.0" \
+    --rm="true" --restart="Never" -- bin/kafka-producer-perf-test.sh \
+    --topic my-topic --record-size 1000 --num-records 100000 --throughput -1 \
+    --producer-props acks=1 bootstrap.servers=my-cluster-kafka-bootstrap:9092
+
+kubectl exec -it my-cluster-kafka-0 -c kafka -- \
+    bin/kafka-console-consumer.sh --bootstrap-server :9092 \
+    --topic my-topic --group my-group --from-beginning --timeout-ms 15000
+
+# save consumer group offsets
+kubectl exec -it my-cluster-kafka-0 -c kafka -- \
+    bin/kafka-consumer-groups.sh --bootstrap-server :9092 \
+    --group my-group --describe > /tmp/offsets.txt
+
+# send additional 12345 messages
+kubectl run kafka-producer-perf-test -it \
+    --image="quay.io/strimzi/kafka:latest-kafka-2.6.0" \
+    --rm="true" --restart="Never" -- bin/kafka-producer-perf-test.sh \
+    --topic my-topic --record-size 1000 --num-records 12345 --throughput -1 \
+    --producer-props acks=1 bootstrap.servers=my-cluster-kafka-bootstrap:9092
+
+# run backup procedure
+./tools/cold-backup/run.sh -b -n myproject -c my-cluster -t /tmp/my-cluster.zip -m custom-cm
+
+# delete the namespace and restore
+kubectl delete ns myproject
+kubectl create ns myproject
+./tools/cold-backup/run.sh -r -n myproject -c my-cluster -s /tmp/my-cluster.zip
+
+# deploy the operator and wait for provisionig
+kubectl create -f ./install/cluster-operator
```

Comment:
Who and where installs the Kafka custom resource? I assume the backup/restore tool does it? You should then probably install the operator first before running the restore. In case you would be restoring on a different cluster, you might not have the CRDs installed until this step and that would fail the restore.

Reply:
Yes, I thought about this, but in practice doing like this ends with zookeeper error on TLS certificates. More details later.

Reply:
This is the issue with switching these two steps in the ST. It seem to work when doing it manually. Will do some more tests.

```
2021-03-16 15:26:05,574 WARN Exception caught (org.apache.zookeeper.server.NettyServerCnxnFactory) [nioEventLoopGroup-7-2]
io.netty.handler.codec.DecoderException: javax.net.ssl.SSLHandshakeException: PKIX path validation failed: java.security.cert.CertPathValidatorException: signature check failed
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:471)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:276)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:163)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:714)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:650)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:576)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:493)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

Reply:
Ok I fixed that and switched the operator deploy before restore procedure.